### PR TITLE
build: upgrade to go1.17

### PR DESCRIPTION
**Pull request by docmerlin**

### CODE REVIEW

**Review by lesam — APPROVED** [review]
lgtm

**Comment by docmerlin** [comment]
I'm putting this on hold as it breaks cross builds for some reason.

File: Dockerfile_build_ubuntu64
Comment (danxmoran) [inline Dockerfile_build_ubuntu64:2]: NOTE: This is from a branch in the `edge` repo, pushed to confirm it's working as expected

File: .circleci/config.yml
Comment (lesam) [inline .circleci/config.yml:71]: I think you should be able to request a specific VM image and avoid the docker upgrade

File: build.py
Comment (danxmoran) [inline build.py:234]: Might want to add this back, but under a flag (it annoyed me locally but is useful in CI to see that it's not frozen)

File: .circleci/config.yml
Comment (danxmoran) [reply .circleci/config.yml:71]: :+1: the image I've pinned in the `executors` definition is the same as used in influxdb 2.x / influx-cli, and I remember its pre-installed Docker being pretty old. I'll check if there's a newer image out that we can try.

File: Dockerfile_build_ubuntu64
Comment (danxmoran) [reply Dockerfile_build_ubuntu64:2]: Will need updated after influxdata/edge#109 merges

File: build.py
Comment (lesam) [reply build.py:234]: I think we do want it back for CI, there I want all the info I can get after the fact

File: build.py
Comment (lesam) [inline build.py:207]: Looks like `ForCompiler` in that package is module-aware - can we have that added to the comment here so when we fix it we know where to look?

File: udf/agent/server.go
Comment (lesam) [inline udf/agent/server.go:77]: Why?

Comment (danxmoran) [reply udf/agent/server.go:77]: It failed `go vet`, reasoning explained in the release notes [here](https://golang.org/doc/go1.17#vet).

File: server/server_const_windows_test.go
Comment (lesam) [inline server/server_const_windows_test.go:1]: Why the changed build tag format? Do we need both?

Comment (danxmoran) [reply server/server_const_windows_test.go:1]: Yes, it's auto-generated by `go fmt` on 1.17 and `go vet` complains if you try to delete the old form. Notes start [here](https://golang.org/doc/go1.17#gofmt).

File: Dockerfile_build_ubuntu64
Comment (lesam) [inline Dockerfile_build_ubuntu64:1]: Excited about every change making this smaller. Looks like all the rest of this could be pushed up to the edge repo too - maybe in a follow-up PR?

Comment (danxmoran) [reply Dockerfile_build_ubuntu64:1]: For now, I'll delete all the things from the `apt-get install` that are already covered by the cross-builder.

**Review by lesam — APPROVED** [review]
LGTM, but would be nice to have the ForCompiler comment

**Pull request merged**
